crates/sqlexec/src/planner/errors.rs

```diff
@@ -94,6 +94,20 @@ impl From<crate::errors::ExecError> for PlanError {
     }
 }
 
+macro_rules! impl_from_dispatch_variant {
+    ($FromType:ty) => {
+        impl From<$FromType> for PlanError {
+            #[inline]
+            fn from(err: $FromType) -> Self {
+                PlanError::Dispatch(crate::dispatch::DispatchError::from(err))
+            }
+        }
+    };
+}
+impl_from_dispatch_variant!(datasources::lake::delta::errors::DeltaError);
+impl_from_dispatch_variant!(datasources::lake::iceberg::errors::IcebergError);
+impl_from_dispatch_variant!(datasources::object_store::errors::ObjectStoreSourceError);
+
```

Comment:
Feels like there might be a cleaner way to do this, but no luck with a generic from along the lines 
```rust
impl<E: Into<crate::dispatch::DispatchError>> From<E> for PlanError {
    fn from(err: E) -> Self {
        PlanError::Dispatch(crate::dispatch::DispatchError::from(err))
    }
}
```

Since this leads to "conflicting implementations of trait `From<DispatchError>` for type `PlanError`"